corehq/apps/users/tests/test_quickcache_race.py

```diff
@@ -0,0 +1,59 @@
+import threading
+
+from django.test import SimpleTestCase
+from corehq.util.quickcache import quickcache
+from time import sleep
+
+
+@quickcache(['key'])
+def generate_value(key, call_tracker, duration=0):
+    call_tracker['calls'] = call_tracker['calls'] + 1
+    sleep(duration)
+    return f'hello {key}'
+
+
+def set_value(call_tracker, duration):
+    generate_value('world', call_tracker, duration)
+
+
+def clear_cache(delay):
+    sleep(delay)
+    generate_value.clear('world', None)
+
+
+class TestQuickcacheRace(SimpleTestCase):
+    def setUp(self):
```

Comment:
If the tests are moved to quickcache but left in a failing state this might be a reasonable addition (with imports moved to top of module, of course)

```suggestion
    def setUp(self):
        import os
        from unittest import SkipTest
        if not os.environ.get("QUICKCACHE_TEST_RACES"):
            raise SkipTest("race condition test skipped because it is slow and has not yet been fixed")
```